pymodbus/repl/client/main.py

```diff
@@ -342,7 +332,7 @@ def tcp(ctx, host, port, framer):
     "PARITY_NONE, PARITY_EVEN, PARITY_ODD PARITY_MARK, "
     'PARITY_SPACE. Default to "N"',
     default="N",
-    type=CaseInsenstiveChoice(["N", "E", "O", "M", "S"]),
+    type=click.Choice(["N", "E", "O", "M", "S"], case_sensitive=False),
```

Comment:
this will automatically normalize to the values in the list (upper case) so we can kill `CaseInsensitiveChoice` class entirely

```
alex@antorak:~/git/pymodbus$ pymodbus.console serial --port /dev/blah --parity e
> client.get_parity
"E"
```